### PR TITLE
Emit Spark user metrics in the correct scope.

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/metrics/SparkUserMetrics.java [hunk 1]
```diff
@@ -34,24 +35,24 @@
  */
 public final class SparkUserMetrics implements Metrics, Externalizable {
 
-  private final MetricsContext metricsContext;
+  private final Metrics delegate;
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/metrics/SparkUserMetrics.java]: For my info, why was this change needed?

File: cdap-unit-test/src/test/java/co/cask/cdap/spark/metrics/TestSparkMetricsIntegrationApp.java [hunk 2]
```diff
@@ -21,9 +21,12 @@
 import co.cask.cdap.api.spark.AbstractSpark;
 import co.cask.cdap.api.spark.JavaSparkProgram;
 import co.cask.cdap.api.spark.SparkContext;
+import com.google.common.collect.ImmutableList;
```
Comment (bdmogal) [inline cdap-unit-test/src/test/java/co/cask/cdap/spark/metrics/TestSparkMetricsIntegrationApp.java]: unused imports? This and line 28-29?

**Comment by bdmogal** [comment]
Just a couple of questions. Code LGTM.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/metrics/SparkUserMetrics.java (hunk 1 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/metrics/SparkUserMetrics.java]: Oh, nvm, this makes the metrics be emitted in user context. :+1:

**Comment by chtyim** [comment]
Fixed unused imports and build passed: http://builds.cask.co/browse/CDAP-DUT3112-5

**Comment by chtyim** [comment]
squashed and merging.
